### ISSUE TITLE
`Fetchers\Post`: Accept slugs/paths in addition to IDs

### ISSUE THREAD

**Issue by iandunn** [wp-cli/wp-cli#5842]
## Feature Request

It'd be convenient to be able to `wp post get foo` instead of having to manually look up the ID and then run `wp post get 32425`.

**Is your feature request directly related to a specific, existing command?**

The user interacts with `post get` in the entities repo, but it seems like it'd be implemented in `Fetchers\Post` in this repo.

- [x] Yes, I reviewed the [contribution guidelines](https://make.wordpress.org/cli/handbook/contributing/).

**Describe your use case and the problem you are facing**

I often know a URL like `https://example.org/my-cpt/foo/`, and want to run something like `wp post get 32523` to get some info about the post. In order to do that, I have to:

1. Switch to a browser
1. Login to `wp-admin`
2. Visit the URL
3. Click `Edit` in the admin toolbar
4. Copy the ID out of the URL
5. Switch back to terminal
6. Paste the ID

If feels like that could be simplified.

**Describe the solution you'd like**

`Fetchers\User`, `term get`, etc already allow passing slugs in addition to IDs. It'd be much more convenient if I just could type `wp post get my-cpt/foo`. Maybe something like:

```php
if ( is_numeric( $identifier ) ) {
    $post = get_post( $identifier );
} else {
    $cpt_slug = 'my_cpt'; // get by extracting first part of path and mapping to a rewrite rule?
    $post     = get_page_by_path( 'foo', OBJECT, $cpt_slug );
}
```

There'd need to be extra logic to handle other URL patterns like archives etc, but there may be a way to leverage `parse_request()` or something to do most of that work for us.

Alternatively, maybe I could pass a full `--URL`, and then leave ID empty, like:

`wp post get --url=https://example.org/my-cpt/foo`

...and the bootstrap process would populate the global `$post` based on that?

**Comment by danielbachhuber** [comment]
Thanks for the suggestion, @iandunn.

The underlying challenge here is that posts can have the same slug if they're of different post types. 

Additionally, "[composability is always a good idea](https://github.com/wp-cli/wp-cli/issues/5487#issuecomment-759079667)" kicks in.

I wouldn't want to introduce new behavior for `wp post get`, but we could introduce a new command like `wp post url-to-id` for it. Thoughts?

**Comment by iandunn** [comment]
Fair point about composability. `url-to-id` seems like a good solution 👍🏻
